### DISCUSSION TITLE
How do you automatically get your headers

### DISCUSSION THREAD

**Discussion by PoIygon** [theforumhelpers/theforumhelpers.github.io#198] [Q&A]
I was wondering because I’m trying to do it for my site

**Comment by Accio1** [comment]
Each page has an empty header and footer element, and loads an imports.js file with `<script src="resources/imports.js" onload='reference("")'></script>`. This imports.js file sets the innerHTML of the header and footer to match the headerContent and footerContent constants. This means that by changing those constants, you can update the header and footer on each page without having to change it more than once. If you have a specific question about how you could implement this on your own site, feel free to reply and ask!

**Reply by PoIygon** [reply]
Is there an easier way to do this?

**Reply by Accio1** [reply]
Aside from using a JS framework like Vue, this is the easiest way I cold think of. It isn't that difficult once you get everything set up.

**Comment by aidan-crockett** [comment]
With a little bit of extra work you can set it up so that you only need to include the `<script>` tag - `<script src="resources/index.js">`
